### ISSUE TITLE
how to draw arrow on line layer?

### ISSUE THREAD

**Issue by BangDori** [mapbox/mapbox-gl-js#12891]
**mapbox-gl-js version**:

    "react-map-gl": "^7.0.21"

### Question

![image](https://github.com/mapbox/mapbox-gl-js/assets/44726494/d19c5388-df53-4dfa-8dfb-52bd5b3b02db)

I'm using react-map-gl on web-application for capstone project. I want to draw arrow direction on center between two points. but, I don't know draw arrow direction on center.

how to draw arrow direction like under the image...?

![1234](https://github.com/mapbox/mapbox-gl-js/assets/44726494/eaa64cc8-e78b-495a-aabd-99ac5eec3846)

**Comment by BangDori** [comment]
I solved this problem using **simbol type**.

1. upload arrow.svg in mapbox studio.
2. mapbox studio publishing
3. apply symbol Layer as under the code.
   ```
   <Layer
      id='arrowLayer'
      type='symbol'
      source='polylineLayer'
      layout={{
         'symbol-placement': 'line',
         'symbol-spacing': 100,
         'icon-image': 'arrow1',
         'icon-size': 0.75,
      }}
   />
   ```
4. wait a minute....
5. Success !
![image](https://github.com/mapbox/mapbox-gl-js/assets/44726494/c0901931-7dbb-492c-b317-3332f4f25c90)



### I have a question. I don't know why it takes so long to be applied.

**Comment by stepankuzmin** [comment]
Hi @BangDori,

Thank you for using mapbox-gl-js. The symbol layer should work for that case. Could you please elaborate more on what is the issue at the moment?

Also, for future "how do I" questions like this, you can refer to our [help documentation](https://www.mapbox.com/help/), and if you can't find the answer there, [contact support](https://www.mapbox.com/contact/). This issue tracker is for reporting bugs and feature requests.

You might also consider posting your question to https://stackoverflow.com/questions/tagged/mapbox-gl-js to ask the community for help.

**Comment by BangDori** [comment]
Thank you for reading this issue @stepankuzmin.

What I wanted to ask was that the SVG file I uploaded to Mapbox Studio wasn't immediately recognized by the web server.

To elaborate further:

I successfully uploaded the SVG file to Mapbox Studio and publish Mapbox Studio.

then, after setting the `icon-name` of the symbol layer to match the name of the SVG file registered in Mapbox Studio, I restarted the web server. 

However, it didn't take effect immediately. It took approximately 30 minutes before I could confirm that the symbol layer's icon was applied correctly.
